### ISSUE TITLE
gperftools 2.6.1 is not compatible with gpertools 2.4.1

### ISSUE THREAD

**Issue by vonnyfly** [gperftools/gperftools#1006]
This is a simplest code to reproduce this issue.
Build this code with gcc7.2 on centos 7.
```
 $ cat a.cc                                                                                                                                                                   
struct A {
};

int main() {
    A *a = new A();
    delete a;
}

$ . /opt/rh/devtoolset-7/enable
$ g++ --std=c++17 -ltcmalloc a.cc
$ ./a.out
$

Copy a.out to production machine with gperftools 2.4.

[root@node69 17:45:54 ~]$./a.out
./a.out: symbol lookup error: ./a.out: undefined symbol: _ZdlPvm

On dev machine:
 $ rpm -qa | grep gperftools                                                                                                                                                  
gperftools-devel-2.6.1-1.el7.x86_64
gperftools-libs-2.6.1-1.el7.x86_64
gperftools-2.6.1-1.el7.x86_64

On production machine:
[root@node69 17:46:21 ~]$rpm -qa | grep gperftools
gperftools-devel-2.4-8.el7.x86_64
gperftools-libs-2.4-8.el7.x86_64
gperftools-2.4-8.el7.x86_64


 $ nm a.out
0000000000601034 B __bss_start
0000000000601034 b completed.6930
0000000000601030 D __data_start
0000000000601030 W data_start
00000000004005a0 t deregister_tm_clones
0000000000400610 t __do_global_dtors_aux
0000000000600de0 t __do_global_dtors_aux_fini_array_entry
0000000000400708 R __dso_handle
0000000000600de8 d _DYNAMIC
0000000000601034 D _edata
0000000000601038 B _end
00000000004006f4 T _fini
0000000000400640 t frame_dummy
0000000000600dd8 t __frame_dummy_init_array_entry
0000000000400858 r __FRAME_END__
0000000000601000 d _GLOBAL_OFFSET_TABLE_
                 w __gmon_start__
0000000000400710 r __GNU_EH_FRAME_HDR
00000000004004f8 T _init
0000000000600de0 t __init_array_end
0000000000600dd8 t __init_array_start
0000000000400700 R _IO_stdin_used
00000000004006f0 T __libc_csu_fini
0000000000400680 T __libc_csu_init
                 U __libc_start_main@@GLIBC_2.2.5
0000000000400647 T main
00000000004005d0 t register_tm_clones
0000000000400570 T _start
0000000000601038 D __TMC_END__
                 U _ZdlPvm
                 U _Znwm

On dev machine：
 $ nm -D /usr/lib64/libtcmalloc.so | grep dlPvm
0000000000035f40 T _ZdlPvm

On production machine：
[root@node69 17:54:19 ~]$nm -D /usr/lib64/libtcmalloc.so | grep dlPvm
[root@node69 17:54:24 ~]$
```
Any one could explain this issue?
I suspect the gperftools-2.6.1-1.el7.x86_64 was built with cpp17, but I could't verify it.
Millions of thanks

**Comment by xiaoyur347** [comment]
It sounds an issue for overaligned type new in 2.6.1.  
https://github.com/gperftools/gperftools/commit/7efb3ecf37d88edf9cf9a43efb89b425eaf81d5e
```
$c++filt _ZdlPvm
operator delete(void*, unsigned long)
```
It's fixed in 2.6.2. Please test the newer version to see the problem still exists.

**Comment by alk** [comment]
Real issue is trying to use program linked against gperftools 2.6, on system with gperftools 2.4. That isn't going to work.

If you link tcmalloc into your program statically, then it should be fine.
